src/Components/Web.JS/src/FileDownloader.ts

```diff
@@ -0,0 +1,14 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+export async function downloadFile(data: any, fileName: string) {
+    const arrayBuffer = await data.arrayBuffer();
+    const blob = new Blob([arrayBuffer]);
+    const url = URL.createObjectURL(blob);
+    const anchorElement = document.createElement('a');
+    anchorElement.href = url;
+    anchorElement.download = fileName;
+    anchorElement.click();
+    anchorElement.remove();
```

Comment:
I think the `anchorElement` needs to be added to the DOM e.g.
```js
document.body.appendChild(anchorElement);
```
to make it work in Firefox. At least that is what I have in my own custom implementation.
Otherwise the removal isn't needed I guess.